api/v1beta1/event_helpers.go

```diff
@@ -0,0 +1,465 @@
+package v1beta1
+
+import (
+	"errors"
+	"fmt"
+)
+
+// EventDataType constraint for type-safe data extraction from Event.Data
+type EventDataType interface {
+	string | int32 | int64 | uint32 | uint64 | []byte | bool
+}
+
+// GetData extracts a value from Event.Data by name with type safety.
+// Returns the value and true if found and type-compatible,
+// or zero value and false if not found or type mismatch.
+// Use for optional fields where missing data is acceptable.
+func GetData[T EventDataType](event *Event, name string) (T, bool) {
```

Comment:
Adding it here, but valid for the whole fail, we need tests for all methods here!